CoordinatorTests.testUnresponsiveLeaderDetectedEventually reproducible failure:

This fail in one PR but I seem to be able to reproduce it in master.

**Build scan**:

https://gradle-enterprise.elastic.co/s/d5pmmsn5fz3bi

**Repro line**:

```
 ./gradlew ':server:test' --tests "org.elasticsearch.cluster.coordination.CoordinatorTests.testUnresponsiveLeaderDetectedEventually" -Dtests.seed=5C32928A3DFC7F15 -Dtests.security.manager=true -Dtests.locale=bg-BG -Dtests.timezone=Mexico/General -Druntime.java=11

```
**Reproduces locally?**:

yes

**Applicable branches**:

Only tried on master

**Failure history**:

Seems not to have fail in a long time.

**Failure excerpt**:


```
  1> [2020-10-20T04:56:15,000][INFO ][o.e.c.c.CoordinatorTests ] [testUnresponsiveLeaderDetectedEventually] after test
  2> REPRODUCE WITH: ./gradlew ':server:test' --tests "org.elasticsearch.cluster.coordination.CoordinatorTests.testUnresponsiveLeaderDetectedEventually" -Dtests.seed=5C32928A3DFC7F15 -Dtests.security.manager=true -Dtests.locale=bg-BG -Dtests.timezone=Mexico/General -Druntime.java=11
  2> java.lang.AssertionError: node1 has applied its state 
    Expected: <10L>
         but: was <9L>
        at __randomizedtesting.SeedInfo.seed([5C32928A3DFC7F15:265B95CAD8EEBC04]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.elasticsearch.cluster.coordination.AbstractCoordinatorTestCase$Cluster.stabilise(AbstractCoordinatorTestCase.java:537)
        at org.elasticsearch.cluster.coordination.CoordinatorTests.testUnresponsiveLeaderDetectedEventually(CoordinatorTests.java:532)
  2> NOTE: leaving temporary files on disk at: /Users/ivera/forks/elasticsearch-fork/server/build/testrun/test/temp/org.elasticsearch.cluster.coordination.CoordinatorTests_5C32928A3DFC7F15-003
  2> NOTE: test params are: codec=Asserting(Lucene87): {}, docValues:{}, maxPointsInLeafNode=984, maxMBSortInHeap=7.724307660680699, sim=Asserting(RandomSimilarity(queryNorm=false): {}), locale=bg-BG, timezone=Mexico/General
  2> NOTE: Mac OS X 10.15.6 x86_64/N/A 11.0.8 (64-bit)/cpus=12,threads=1,free=506481344,total=536870912
  2> NOTE: All tests run in this JVM: [CoordinatorTests]
```

--- a/server/src/test/java/org/elasticsearch/cluster/coordination/CoordinatorTests.java
+++ b/server/src/test/java/org/elasticsearch/cluster/coordination/CoordinatorTests.java
@@ -544,6 +544,9 @@ public class CoordinatorTests extends AbstractCoordinatorTestCase {
                     // so we should take into account those cases to ensure that the
                     // cluster stabilises over time. See #63918 for a really messy scenario.
                     + DEFAULT_ELECTION_DELAY
+                    // additionally take into account that publications might take longer
+                    // until the new leader detects that the old leader is unresponsive
+                    + defaultMillis(PUBLISH_TIMEOUT_SETTING)
 
                     // then wait for both of:
                     + Math.max(

--- a/server/src/test/java/org/elasticsearch/cluster/coordination/CoordinatorTests.java
+++ b/server/src/test/java/org/elasticsearch/cluster/coordination/CoordinatorTests.java
@@ -539,6 +539,11 @@ public class CoordinatorTests extends AbstractCoordinatorTestCase {
                     + defaultMillis(PUBLISH_TIMEOUT_SETTING)
                     // there might be a term bump causing another election
                     + DEFAULT_ELECTION_DELAY
+                    // in clusters with 5 nodes the chances of concurrent elections
+                    // increase, meaning that it takes longer to get a leader elected
+                    // so we should take into account those cases to ensure that the
+                    // cluster stabilises over time. See #63918 for a really messy scenario.
+                    + DEFAULT_ELECTION_DELAY
 
                     // then wait for both of:
                     + Math.max(

